extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -230,6 +176,10 @@ public void handle(AsyncResult<Void> event) {
             } catch (InterruptedException e) {
                 throw new RuntimeException(e);
             }
+            synchronized (devModeThreads) {
```

Comment:
Hello Stuart, is there a chance that the threads and classloader will leak if the previous try block throws an exception? In other words, should we use a `finally` block here to clear the collection and the classloader?

Reply:
In theory this could happen, I will change it.